lib/link-to.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  Directive,
+  HostBinding,
+  HostListener,
+  Input
+} from  'angular2/core';
+import { Location, normalizeQuery } from './location';
+
+/**
+ * The LinkTo directive links to routes in your app
+ *
+ * Links are pushed to the `Location` service to trigger a route change.
+ * Query params can be represented as an object or a string of names/values
+ *
+ * <a [linkTo]="/home/page" [queryParams]="{ id: 123 }">Home Page</a>
+ */
+@Directive({ selector: 'linkTo' })
+export class LinkTo {
+  @Input() target: string;
+  @HostBinding('attr.href') linkHref;
+  private _href: string;
+  private _query: string | Object;
+
+  @Input() set linkTo(href: string){
+    this._href = href;
+    this._updateHref();
+  }
+
+  @Input() set queryParams(params: string | Object) {
+    this._query = params;
+    this._updateHref();
+  }
+
+  constructor(private _location: Location) {}
+
+  /**
+   * Handles click events on the associated link
+   * Prevents default action for non-combination click events without a target
+   */
+  @HostListener('click', ['$event'])
+  onClick(event) {
+    if (!this._comboClick(event) && !this.target) {
+      this._location.go(this.linkTo, this.queryParams);
+
+      event.preventDefault();
+    }
+  }
+
+  private _updateHref() {
+    this.linkHref = `${this._href}${normalizeQuery(this._query)}`;
```

Comment:
Rather than normalizing here, let's change `Location.prepareExternalUrl()` to accept both a URL fragment and a query string/object. Then the href should be the external URL.